sematic/resolvers/state_machine_resolver.py

```diff
@@ -230,13 +248,16 @@ def _schedule_future_if_args_resolved(self, future: AbstractFuture) -> None:
             self._execute_future(future)
 
     def _execute_future(self, future: AbstractFuture) -> None:
-        self._future_will_schedule(future)
         if future.props.inline:
-            logger.info("Running inline {}".format(future.calculator))
+            self._future_will_schedule(future)
+            logger.info("Running inline %s", future.calculator)
             self._run_inline(future)
-        else:
-            logger.info("Scheduling {}".format(future.calculator))
+        elif self._can_schedule_future(future):
+            self._future_will_schedule(future)
+            logger.info("Scheduling %s", future.calculator)
             self._schedule_future(future)
+        else:
+            logger.info("Currently not scheduling %s", future.calculator)
```

Comment:
We can rationalize this more by including an `inline` check in `_can_schedule_future` and avoid the double call to `_future_will_schedule`:
```python
if not self._can_schedule_future(future):
    logger.info("Currently not scheduling %s", future.calculator)
    return

if future.props.inline:
    logger.info("Running inline %s", future.calculator)
    self._run_inline(future)
else: 
    logger.info("Scheduling %s", future.calculator)
    self._schedule_future(future)
```

Reply:
Done.